pkg/stub/handler.go

```diff
@@ -187,12 +189,23 @@ func (h *Handler) StoreCurrentValidConfig(condition v1alpha1.SamplesResourceCond
 		// 4.0 testing showed that we were getting empty ConfigMaps
 		cm = nil
 	}
+	//TODO our create default samplesresource logic attempls to create the config map,
+	// but for now we have redundancy here in case it still does not exists
+	// Is this valid ??
```

Comment:
It is probably not valid.  It means we've created samples, but we don't have a record of the configuration that lead to those samples.  If we don't have that record, we can't validate changes to the configuration, which means we can't ever have another known valid configuration.

I would say if the configmap does not exist, and the samples do exist, we are in a broken state that requires user/admin intervention to reconcile.

Reply:
sounds good ... will propagate as an error